src/node_i18n.cc

```diff
@@ -524,14 +536,14 @@ void ConverterObject::Decode(const FunctionCallbackInfo<Value>& args) {
     Local<Value> ret;
     if (encoded.ToLocal(&ret)) {
       args.GetReturnValue().Set(ret);
-    } else {
-      args.GetReturnValue().Set(error);
+      return;
     }
-
-    return;
   }
 
-  args.GetReturnValue().Set(status);
+  node::THROW_ERR_ENCODING_INVALID_ENCODED_DATA(
+      env->isolate(),
+      "The encoded data was not valid for encoding %s",
+      *node::Utf8Value(env->isolate(), from_encoding));
```

Comment:
Can you assign the Utf8Value to a local variable?

Dereferencing a temporary is normally safe within an expression ("full expression" the C++ spec calls it) but since you're passing it to a macro, it's not obvious what the full expression is.

Reply:
This was supposed to be in a different pull request, but due to the force-push, it seems like it's included in this pull request. I just force-pushed it.